### PR TITLE
add from_array and from_pil helper functions to Image type

### PR DESCRIPTION
**Pull request by eddyxu**

### CODE REVIEW

**Comment by eddyxu** [comment]
Also fixed a few places to use `pathlib.Path` for connivence.

File: python/rikai/types/vision.py [hunk 1]
```diff
@@ -40,10 +49,61 @@ class Image(ToNumpy, Asset, Displayable):
...
+        Parameters
+        ----------
+        img : :py:class:`PIL.Image`
+            An PIL Image instance
+        uri : str or Path
+            The URI to store the image externally.
+        """
+        parsed = urlparse(normalize_uri(uri))
+        if parsed.scheme == "file":
+            img.save(uri)
```
Comment (changhiskhan) [inline python/rikai/types/vision.py:99]: I *think* this should also work directly for dbfs

File: python/rikai/types/vision.py [hunk 2]
```diff
@@ -40,10 +49,61 @@ class Image(ToNumpy, Asset, Displayable):
...
+        """Create an image in memory from a :py:class:`PIL.Image`.
+
+        Parameters
+        ----------
+        img : :py:class:`PIL.Image`
+            An PIL Image instance
+        uri : str or Path
+            The URI to store the image externally.
+        """
+        parsed = urlparse(normalize_uri(uri))
```
Comment (changhiskhan) [inline python/rikai/types/vision.py:97]: This is something we should probably standardize (there are lots of places whether we have this if this scheme do x else y)

**Review by changhiskhan — APPROVED** [review]

File: python/rikai/types/vision.py [hunk 3]
```diff
@@ -40,10 +49,61 @@ class Image(ToNumpy, Asset, Displayable):
...
+        if parsed.scheme == "file":
+            img.save(uri)
+        else:
+            with NamedTemporaryFile() as fobj:
+                img.save(fobj)
+                fobj.flush()
+                copy(fobj.name, uri)
```
[the remark this answers is not in the document]
Comment (eddyxu) [reply python/rikai/types/vision.py:104]: This works via `copy` (for s3 at least today). 

`rikai.io` supposes to handle the logic of handing different cloud storage backends.

File: python/rikai/types/vision.py (hunk 1 above)
Comment (eddyxu) [reply python/rikai/types/vision.py:99]: you meant `PIL` can directly save to `dbfs`?

Comment (eddyxu) [reply python/rikai/types/vision.py:99]: We could standardize to get a `write fobj / stream` from `uri`, and make them works with these third party libs that works with `fobj`.

File: python/rikai/types/vision.py (hunk 2 above)
Comment (eddyxu) [reply python/rikai/types/vision.py:97]: yep, agreed.

**Pull request merged**